app/controllers/teacher_interface/uploads_controller.rb

```diff
@@ -25,15 +25,20 @@ def create
     end
 
     def delete
+      @delete_upload_form = DeleteUploadForm.new
     end
 
     def destroy
-      if ActiveModel::Type::Boolean.new.cast(params.dig(:upload, :confirm))
-        @upload.attachment.purge
-        @upload.destroy!
+      confirm = params.dig(:teacher_interface_delete_upload_form, :confirm)
+      upload = load_upload
+      @delete_upload_form =
+        TeacherInterface::DeleteUploadForm.new(confirm:, upload:)
+      if @delete_upload_form.valid?
+        @delete_upload_form.save! if @delete_upload_form.confirm
```

Comment:
I think this conditional logic should be moved in to the save method, again so it can be unit tested easily where confirm is true and false.